lib/auth/accountrecovery_test.go

```diff
@@ -1060,6 +1060,156 @@ func TestAccountRecoveryFlow(t *testing.T) {
 	}
 }
 
+func TestGetAccountRecoveryToken(t *testing.T) {
+	t.Parallel()
+	srv := newTestTLSServer(t)
+
+	cases := []struct {
+		name     string
+		isExpErr bool
+		req      CreateUserTokenRequest
+	}{
+		{
+			name:     "invalid recovery token type",
+			isExpErr: true,
+			req: CreateUserTokenRequest{
+				TTL:  5 * time.Minute,
+				Type: UserTokenTypeResetPassword,
+			},
+		},
+		{
+			name:     "token not found",
+			isExpErr: true,
+			req: CreateUserTokenRequest{
+				TTL:  5 * time.Minute,
+				Type: "Type-Does-Not-Exist",
```

Comment:
Maybe:

```suggestion
				Type: "unknown-token-type",
```

so it looks less like an impossible situation and more like future proofing?